### PR TITLE
Spike escher directive

### PR DESCRIPTION
**Pull request by suliatis**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -3,7 +3,7 @@ val scalaV = "2.12.4"
 
 name          := "escher-akka-http"
 organization  := "com.emarsys"
-version       := "0.2.1"
+version       := "1.0.0"
```
Comment (miklos-martin) [inline build.sbt]: I would cherish that if the 1.0 version would not constrain me to use spray json. I believe it is only used for the parseBody function, which, I think, could be omitted from this library.

File: src/main/scala/com/emarsys/escher/akka/http/EscherDirectives.scala [hunk 2]
```diff
@@ -36,36 +36,33 @@ trait EscherDirectives extends RequestBuilding with EscherAuthenticator {
...
-    })
-  }
+  def escherAuthenticate(trustedServiceNames: List[String])
+                        (implicit ec: ExecutionContext, mat: Materializer, logger: LoggingAdapter): Directive0 =
+    extract (_.request) map authenticate(trustedServiceNames) flatMap (onComplete(_)) flatMap passOrReject
 
-  def checkForwardedHttps(inner: Route) : Route = (ctx: RequestContext) => {
-    val protocolHeader = ctx.request.headers.find(_.name.toLowerCase == "x-forwarded-proto")
-    val notForwardedHttps = protocolHeader.fold(false)(! _.value().contains("https"))
+  private def authenticate(trustedServiceNames: List[String])
```
Comment (miklos-martin) [inline src/main/scala/com/emarsys/escher/akka/http/EscherDirectives.scala]: I'm curious, do this (and the next def) need to be a `PartialFunction` or is it just more convenient?

File: src/test/scala/com/emarsys/escher/akka/http/EscherHttpDirectiveTest.scala [hunk 3]
```diff
@@ -19,70 +20,67 @@ class EscherHttpDirectiveTest
...
+        status shouldBe OK
       }
     }
-  }
 
-  "escherAuthenticate" should {
-    val service = escherConfig.services.head
-    val (key, secret) = (escherConfig.key(service), escherConfig.secret(service))
-    val host = s"${escherConfig.hostName}:${escherConfig.port}"
+    "signed and forwarded https request" in {
```
Comment (miklos-martin) [inline src/test/scala/com/emarsys/escher/akka/http/EscherHttpDirectiveTest.scala]: I would put a verb in here to make it read nicely in the test output

File: build.sbt (hunk 1 above)
Comment (suliatis) [reply build.sbt]: It seems unnecessary, I removed the spray json dependency.

File: src/test/scala/com/emarsys/escher/akka/http/EscherHttpDirectiveTest.scala (hunk 3 above)
Comment (suliatis) [reply src/test/scala/com/emarsys/escher/akka/http/EscherHttpDirectiveTest.scala]: I totally missed that test. Thanks. Fixed it.

File: src/main/scala/com/emarsys/escher/akka/http/EscherDirectives.scala (hunk 2 above)
Comment (suliatis) [reply src/main/scala/com/emarsys/escher/akka/http/EscherDirectives.scala]: It's not necessary, I just saved the `_ match {` from the beginning of the function definition, also in the first function  it not required to use pattern matching, it just looked more functional than an `if/else`. In the other hand the `PartialFunction[A,B]` looks worse than the `A -> B`. :(

**Pull request merged**